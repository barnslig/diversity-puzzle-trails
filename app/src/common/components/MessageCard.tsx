import { Card, CardContent, Typography } from "@mui/material";
import { FormattedRelativeTime } from "react-intl";
import * as React from "react";

import { Message } from "../types/Message";

type MessageCardProps = {
  message: Message;
};

const MessageCard = ({ message }: MessageCardProps) => {
  return (
    <Card
      sx={{
        position: "relative",
        marginBottom: 2,
      }}
      key={message.id}
    >
      <CardContent>
        <Typography variant="body2">{message.attributes.message}</Typography>
        <Typography
          color="textSecondary"
          sx={{
            position: "absolute",
            bottom: 3,
            right: (theme) => theme.spacing(1),
          }}
          variant="caption"
        >
          <FormattedRelativeTime
            value={
              (new Date(message.attributes.createdAt).getTime() -
                new Date().getTime()) /
              1000
            }
            updateIntervalInSeconds={30}
          />
        </Typography>
      </CardContent>
    </Card>
  );
};

export default MessageCard;
