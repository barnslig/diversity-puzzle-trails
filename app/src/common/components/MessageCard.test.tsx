import * as React from "react";
import render from "../testing/render";

import MessageCard from "./MessageCard";

it("renders correctly", () => {
  const { container } = render(
    <MessageCard
      message={{
        type: "message",
        id: "test-123",
        attributes: {
          createdAt: "2021-05-14T12:58:59.063Z",
          message: "Test 123",
        },
      }}
    />
  );
  expect(container.firstChild).toMatchSnapshot();
});
