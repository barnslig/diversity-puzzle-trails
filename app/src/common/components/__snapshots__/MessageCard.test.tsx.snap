// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1al38ny-MuiPaper-root-MuiCard-root"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      Test 123
    </p>
    <span
      class="MuiTypography-root MuiTypography-caption css-2412zy-MuiTypography-root"
    >
      <div />
    </span>
  </div>
</div>
`;
